start.sh

```diff
@@ -26,4 +36,9 @@ cd ${INPUT_DIRECTORY}
 remote_repo="${INPUT_GITHUB_URL_PROTOCOL}//${GITHUB_ACTOR}:${INPUT_GITHUB_TOKEN}@${INPUT_GITHUB_URL}/${REPOSITORY}.git"
 git config --local --add safe.directory ${INPUT_DIRECTORY}
 
-git push "${remote_repo}" HEAD:${INPUT_BRANCH} --follow-tags $_FORCE_OPTION $_TAGS;
+if ${INPUT_FORCE_WITH_LEASE}; then
+  git push --follow-tags $_FORCE_OPTION $_TAGS;
```

Comment:
Do we need to use a different set of options and arguments here? It seems like overkill to me.

Reply:
Unfortunately, yes. Otherwise, it's not working. The combination of the checkout configuration and the corresponding light and different push command solves the problems.

Reply:
Can you provide some more information like logs it might be helpful to know and document the subtleties of using either option.

Reply:
Still curious as to why we need to couple the behaviour with the checkout action as opposed to using the specified branch and also pushing without the remote_repo which includes the user-defined token ??

Reply:
@jackton1 I fired through a corresponding test setup and ran the two actions with two different versions each. 
Version 1 ⇾ Current setup of the push command with force-with-lease support.
Version 2 ⇾ Modified version with rebuilt push command.

You can clearly see in version 1 that the command is not executed and [rejected](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390058?check_suite_focus=true).

```log
Push to branch refs/pull/11/merge
Pushing to https://github.com/ZPascal/grafana_dashboard_templater.git
POST git-receive-pack (8[9](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390058?check_suite_focus=true#step:4:10)9 bytes)
To https://github.com/ZPascal/grafana_dashboard_templater.git
 ! [remote rejected] HEAD -> refs/pull/11/merge (deny updating a hidden ref)
error: failed to push some refs to 'https://github.com/ZPascal/grafana_dashboard_templater.git'
Error: Invalid exit code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ad-m/github-push-action/cfc[10](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390058?check_suite_focus=true#step:4:11)a6161cb45fdaa7cb39dfb870e1ac7c32d3d/start.js:29:21)
    at ChildProcess.emit (events.js:3[14](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390058?check_suite_focus=true#step:4:15):20)
    at maybeClose (internal/child_process.js:1022:[16](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390058?check_suite_focus=true#step:4:17))
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5) {
  code: 1
}
```

In the modified version 2 everything works as [expected](https://github.com/ZPascal/grafana_dashboard_templater/runs/6923390075?check_suite_focus=true).

If you have further questions or want to see log messages, please let me know.

Reply:
Nope looks good. Thanks for the explanation.

Reply:
@jackton1 Could you please approve it or should I modify more code lines?